### PR TITLE
[LibOS] Extend syscall table to 450 syscalls

### PR DESCRIPTION
**Pull request by yao-ji**
## Description of the changes 

The newly added syscalls are unsupported in Gramine, but now they will be
pretty printed like this: `warning: Unsupported system call clone3`.

## How to test this PR? 

Using syscall SYS_clone3 in the application and it prints `warning: Unsupported system call clone3` instead.

---
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)

### CODE REVIEW

**Review by guzongmin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: 0 of 3 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @yao-ji)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/713#-N5sejVm2Wkhhhvc6pUi:-N5sejVm2Wkhhhvc6pUj:b-v7cfqp) (no related file):*
Can you please merge the three commits into one? The two of them may be used to fix some typos/issues in the first commit. I think they should be combined into one commit. 

___

**Comment by yao-ji** [comment]
> > _[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)_ status: 0 of 3 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @yao-ji)
> 
> _[a discussion](https://reviewable.io/reviews/gramineproject/gramine/713#-N5sejVm2Wkhhhvc6pUi:-N5sejVm2Wkhhhvc6pUj:b-v7cfqp) (no related file):_ Can you please merge the three commits into one? The two of them may be used to fix some typos/issues in the first commit. I think they should be combined into one commit.

Done. Thanks.

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by boryspoplawski — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv, @guzongmin, and @yao-ji)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/gramine/713#-N5swoS60ZLhwGYEwpcJ):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would rephrase to this:
```
[LibOS] Extend syscall table to 435 syscalls

The newly added syscalls are unsupported in Gramine, but now they will be
pretty printed like this: `warning: Unsupported system call clone3`.
```
</blockquote></details>

+1
___
*[`libos/include/arch/x86_64/libos_syscalls.h` line 76 at r2](https://reviewable.io/reviews/gramineproject/gramine/713#-N5tFKw0DXfqjtlcPaNj:-N5tFKw0DXfqjtlcPaNk:bxwof80) ([raw file](https://github.com/gramineproject/gramine/blob/f7fbb868fbf945b2abf25ea13ae8a2c6d695d7b1/libos/include/arch/x86_64/libos_syscalls.h#L76)):*
> ```objc
> #endif
> #ifndef __NR_syscalls
> #define __NR_syscalls 436
> ```

While we are here, please just add everything that is available up to today:
```
#define __NR_close_range 436
#define __NR_openat2 437
#define __NR_pidfd_getfd 438
#define __NR_faccessat2 439
#define __NR_process_madvise 440
#define __NR_epoll_pwait2 441
#define __NR_mount_setattr 442
#define __NR_quotactl_fd 443
#define __NR_landlock_create_ruleset 444
#define __NR_landlock_add_rule 445
#define __NR_landlock_restrict_self 446
#define __NR_memfd_secret 447
#define __NR_process_mrelease 448
#define __NR_futex_waitv 449
#define __NR_set_mempolicy_home_node 450
```

**Comment by boryspoplawski** [comment]
Jenkins, retest Jenkins-Debug-18.04 please (`fdatasync01` from ltp failed, known issue)

**Review by yao-ji — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: 0 of 3 files reviewed, 3 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @boryspoplawski, @dimakuv, and @guzongmin)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/713#-N5swT0iEkWLB1y_su2P:-N5xhI5oBybZgET5uqvB:b5pgfhg) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@yao-ji Welcome!

Please use proper Markdown formatting in your descriptions. In particular, please use backticks instead of double-quotes. So:
```
We have "warning: Unsupported system call 435" ...
  ->
We have `warning: Unsupported system call 435` ...
```
</blockquote></details>

Changed. Thanks.

___
___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/gramine/713#-N5swoS60ZLhwGYEwpcJ):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Done. Thanks.
___
*[`libos/include/arch/x86_64/libos_syscalls.h` line 76 at r2](https://reviewable.io/reviews/gramineproject/gramine/713#-N5tFKw0DXfqjtlcPaNj:-N5xhA4n3zlWKsQFQhgL:boyexbg) ([raw file](https://github.com/gramineproject/gramine/blob/90986f48a23c1652a8e699236349ab1be87a25cd/libos/include/arch/x86_64/libos_syscalls.h#L76)):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

While we are here, please just add everything that is available up to today:
```
#define __NR_close_range 436
#define __NR_openat2 437
#define __NR_pidfd_getfd 438
#define __NR_faccessat2 439
#define __NR_process_madvise 440
#define __NR_epoll_pwait2 441
#define __NR_mount_setattr 442
#define __NR_quotactl_fd 443
#define __NR_landlock_create_ruleset 444
#define __NR_landlock_add_rule 445
#define __NR_landlock_restrict_self 446
#define __NR_memfd_secret 447
#define __NR_process_mrelease 448
#define __NR_futex_waitv 449
#define __NR_set_mempolicy_home_node 450
```
</blockquote></details>

Done. Thanks.

**Review by dimakuv — APPROVED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @boryspoplawski and @guzongmin)

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by boryspoplawski — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @guzongmin and @yao-ji)

___
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N68O49iAHSYoxr88fIt:b-ptvni1) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
> ```C
>     [__NR_process_mrelease]       = (libos_syscall_t)0, // libos_syscall_process_mrelease
>     [__NR_futex_waitv]            = (libos_syscall_t)0, // libos_syscall_futex_waitv
>     [__NR_set_mempolicy_home_node]= (libos_syscall_t)0, // libos_syscall_set_mempolicy_home_node
> ```

Hmm, there are two entries now that do not have space before `=`. I would just remove all the newly added entries - we don't need them. This array has correct size anyway, so it's safe to do

**Comment by yao-ji** [comment]
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N6AxG74A2EupMh5hi8x:b-7at12p) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

Hmm, there are two entries now that do not have space before `=`. I would just remove all the newly added entries - we don't need them. This array has correct size anyway, so it's safe to do
</blockquote></details>

Is it safe not initializing these entries to zero? Since all the other "not used" entries are doing this.

**Review by boryspoplawski — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @guzongmin and @yao-ji)

___
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N6B49rQ4hmZCW48N6Ls:btijho9) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
> Is it safe not initializing these entries to zero?

Yes, they will be implicitly initialized anyway

**Comment by yao-ji** [comment]
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N6B8n-J8mnh1i-HEdf6:boyexbg) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

> Is it safe not initializing these entries to zero?

Yes, they will be implicitly initialized anyway
</blockquote></details>

Done. Thanks.

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @boryspoplawski and @guzongmin)

___
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N6CAUmJ4IFo3eDyLgP6:b-6cv5gp) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
<details><summary><i>Previously, yao-ji wrote…</i></summary><blockquote>

Done. Thanks.
</blockquote></details>

@boryspoplawski Are you sure this is readable? This is a confusing choice -- we're removing (or more correctly, not adding) these entries, but we have other (older) entries still in the table. In 3 months, I will look at this table and will be very surprised by this.

I vote for keeping these newly added "zero" entries, even though they are technically not needed.

If we don't like that there is no space before e.g. `[__NR_set_mempolicy_home_node]` -- well, let's add this one space and re-align all other entries.

**Review by boryspoplawski — DISMISSED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @guzongmin and @mkow)

___
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N6DIkjw6fV7ggNJuHYZ:bnzr154) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@boryspoplawski Are you sure this is readable? This is a confusing choice -- we're removing (or more correctly, not adding) these entries, but we have other (older) entries still in the table. In 3 months, I will look at this table and will be very surprised by this.

I vote for keeping these newly added "zero" entries, even though they are technically not needed.

If we don't like that there is no space before e.g. `[__NR_set_mempolicy_home_node]` -- well, let's add this one space and re-align all other entries.
</blockquote></details>

I have no problem with this. I don't even know why we keep these zero entries at all.
But I also don't mind having them - I'm fine with both options, so I'll defer the decision to you.
cc @mkow

**Comment by yao-ji** [comment]
*[`libos/src/arch/x86_64/libos_table.c` line 379 at r4](https://reviewable.io/reviews/gramineproject/gramine/713#-N68O49iAHSYoxr88fIs:-N6GRZn75CypAYN-PdLc:boyexbg) ([raw file](https://github.com/gramineproject/gramine/blob/12f3ca9d75b79c3be882b823c7942f5e38c4d7da/libos/src/arch/x86_64/libos_table.c#L379)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Looks like among the three of us, we have a preference for having code like:
```
    [__NR_set_mempolicy_home_node] = (libos_syscall_t)0, // libos_syscall_set_mempolicy_home_node
```

@yao-ji Sorry for this back-and-forth, but could you restore your version with explicit initialization to zero? Please also add an additional space before `=` in every line of this file.
</blockquote></details>

Done. Thanks.

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv and @guzongmin)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/713#-N6GUGiy8UqBbKt1YpLY:-N6GUGiy8UqBbKt1YpLZ:b-a7gm24) (no related file):*
@yao-ji Please do not force push in the middle of review. And definitely **never** rebase, unless explicitly asked.
I would recommend reading https://gramine.readthedocs.io/en/stable/devel/contributing.html and https://gramine.readthedocs.io/en/stable/devel/onboarding.html

___

**Comment by boryspoplawski** [comment]
Jenkins, test this please

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 1 unresolved discussion (waiting on @guzongmin)

**Review by mkow — APPROVED** [review]
> Reviewed 2 of 3 files at r3, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 1 unresolved discussion (waiting on @guzongmin)

**Comment by mkow** [comment]
Jenkins, retest this please

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @guzongmin)

**Review by boryspoplawski — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/713)* status: all files reviewed, 1 unresolved discussion, not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @guzongmin)
